src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1874,8 +1884,33 @@ private void CreatePropertiesForModelClass(OpenApiUrlTreeNode currentNode, OpenA
                                     .OfType<CodeProperty>()
                                     .ToArray());
             }
-            else if (schema?.AllOf?.LastOrDefault(static x => x.IsObject()) is OpenApiSchema lastAllOfSchema)
-                CreatePropertiesForModelClass(currentNode, lastAllOfSchema, ns, model);
+            finally
+            {
+                model.PropertiesBuildingDone();
+            }
+        }
+    }
+    private Dictionary<String, OpenApiSchema> CollectAllProperties(OpenApiSchema schema)
+    {
+        Dictionary<String, OpenApiSchema> result = new();
+        if (schema.Properties?.Any() ?? false)
+        {
+            foreach (var property in schema.Properties)
+            {
+                result.Add(property.Key, property.Value);
+            }
+        }
+        if (schema.AllOf?.Any() ?? false)
```

Comment:
I'm not sure why we're also generating properties of parent types here? is it so we can detect collisions? won't that lead to a bunch of duplicated work across the board?

Reply:
This change probably also fixes the second test in #2769